app/generate.js

```diff
@@ -97,8 +99,24 @@ function generatePages(renderedComponents) {
       page.conclusionHtml = '';
     }
 
-    page.header = header;
-    page.footer = footer;
+    if (!page.header) {
+      page.header = defaultHeader;
+    } else {
+      if (!page.header.logo) {
+        throw new Error('Header must have logo');
+      }
+      page.header = headerTemplate({
+        url: page.header.logo.url,
+        description: page.header.logoDescription,
+        tagline: page.header.tagline,
+      });
+    }
+
+    if (!page.footer) {
+      page.footer = defaultFooter;
+    } else {
+      page.footer = footerTemplate({ contact: page.footer.email });
```

Comment:
Should be `page.footer.html` here?

Reply:
Well, you _can_ just resolve the conversation - but that doesn't make me think I was wrong? If you comment on why, I have a better chance to understand.

Reply:
Check updated pullrequest